### PR TITLE
Allow any pytest version

### PR DESCRIPTION
**Pull request by andrewfullard**
### :pencil: Description

**Type:** :beetle: `bugfix` | :vertical_traffic_light: `testing` 

Removes reliance on deprecated pytest.config. Doesn't actually work properly though, certainly not for cmfgen tests for some reason.

### :pushpin: Resources

https://docs.pytest.org/en/7.1.x/example/simple.html#request-example

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by andrewfullard** [comment]
Looks like pytest-arraydiff isn't quite working as expected.

File: carsus/conftest.py [hunk 1]
```diff
@@ -72,6 +72,18 @@ def pytest_addoption(parser):
...
+def withtestdb(request):
+    return request.config.getoption("--test-db")
+
+@pytest.fixture
+def withrefdata(request):
+    return request.config.getoption("--refdata")
+
+@pytest.fixture
+def runslow(request):
+    return request.config.getoption("--runslow")
```
Comment (atharva-2001) [inline carsus/conftest.py:86]: I read somewhere that importing fixtures for contfest is generally not a good practice. Should we be moving markers to`pytest.ini`? What do you think?

Comment (andrewfullard) [reply carsus/conftest.py:86]: Well these aren't working very well regardless. For the cmfgen tests they are not correctly skipping if `--refdata` is not part of the pytest command. If you want to take a look and modify, please do so.

Comment (atharva-2001) [reply carsus/conftest.py:86]: I would look into this. Should I create a separate PR then?

Comment (andrewfullard) [reply carsus/conftest.py:86]: I think you should just modify this PR.

Comment (atharva-2001) [reply carsus/conftest.py:86]: sure

**Comment by andrewfullard** [comment]
@shreyas3156 can you investigate why the NNDC test in the notebook is failing? It seems to only be a very tiny difference.

**Pull request merged**